Align OSD waypoint number display with geospatial waypoints

breadoven opened the pull request:
Waypoint missions that include nongeospatial waypoints, JUMP, HEADING, POI, display WP numbers in the OSD that follow the numeric sequence of all WPs in the mission rather than just the geospatial WPs which would seem a more intuitive expectation from a user point of view. This means the displayed WP numbers can appear inconsistent and increment by more than 1 when nongeo WPs are passed.

This PR fixes this by realigning the WP numbers displayed in the OSD with just the geospatial WPs and also adds a geospatial WP total count.

Tested in flight and works as expected for fixed wing and multirotor.

breadoven committed:
Display only geospatial WPs on OSD

Fix to ensure only geospatial WPs are displayed on OSD with sequential numbering maintained around nongeo WPs.

breadoven committed:
Update Method

Change from using WP.flag to WP.p3

breadoven committed:
Update navigation.c

breadoven committed:
Update osd.c

breadoven committed:
Fix white space

breadoven committed:
Change Index List

Change from using WP .P3 to dedicated geospatial WP list list array

avsaase commented on src/main/io/osd.c:
```diff
@@ -3378,7 +3384,7 @@ textAttributes_t osdGetSystemMessage(char *buff, size_t buff_size, bool isCenter
                     if (NAV_Status.state == MW_NAV_STATE_WP_ENROUTE) {
                         // Countdown display for remaining Waypoints
-                        tfp_sprintf(messageBuf, "TO WP %u/%u", posControl.activeWaypointIndex + 1, posControl.waypointCount);
+                        tfp_sprintf(messageBuf, "TO WP %u/%u (%3u M)", getGeoWaypointNumber(posControl.activeWaypointIndex), posControl.geoWaypointCount, (int)(posControl.wpDistance / 100));
```
Would it fit to add the distance unit? For WP distance above 1km, it could show the distance in xx.x km format. Our American friends would probably appreciate empirical units.

breadoven replied:
Ah good point. I'll need to check that.

breadoven committed:
Change of method

Change to remove array for storing geospatial WP indices. Uses iterative method to determine geospatial WP numbering instead.

breadoven committed:
Update navigation_private.h

breadoven committed:
Update osd.c
